### PR TITLE
OPENEUROPA-372: Allow custom commands to define options

### PR DESCRIPTION
**Pull request by claudiu-cristea**
## Problem

As the README states, the runner config can be controlled either by arguments and options passed on the command line or by configuring the values in the `runner.yml.dist` and `runner.yml` files. However, passing options to the command line is true for regular annotated commands. But this library introduces the concept of _dynamic commands_. And for such commands there's no way to define run-time options. How to override in CI a specific config value for the `drupal:site-setup`, for instance, which is defined dynamically?

## Proposal

TL;DR Allow options that can be bound to config entries for custom commands.

Allow defining options for custom commands. Such options could be bound to a specific config entry to overwrite it. See the README and the test added in this PR for details.

Also, custom commands that have subsequent tasks, are exposing also their options, so that calling a custom command with options, when is executed, will pass the options to subsequent tasks.

### CODE REVIEW

**Comment by ademarco** [comment]
@claudiu-cristea we have opened an issue about this on our board.

**Comment by claudiu-cristea** [comment]
I changed the scope of this issue. Defining custom commands options is just enough to achieve the CI problem.

**Comment by ademarco** [comment]
@claudiu-cristea this is a very interesting addition, would you maybe have time to rebase it on the latest change?

**Comment by geek-merlin** [comment]
Rebased #157 now worksforme.
